### PR TITLE
add user repository

### LINKED ISSUE #108: Add user repository
## Feature Title

Add an actual user repository code.

## User Stories

None

## Acceptance Criteria

- [x] find
- [x] create
- [x] update
- [x] delete

## Technical Details

- API Changes: None
- Database Changes: Add Dynamo DB
- Dependencies: Add AWS SKD

## Additional Context

Add DynamDB client

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Added the user repository.

## Review points

This PR uses the repository error, which was created in #145.

### CODE REVIEW

File: backend/main/src/Shared/Adapters/DynamoClient.ts
Comment (taniguchiiqqq) [inline backend/main/src/Shared/Adapters/DynamoClient.ts:1]: The same file exist in `backend/main/src/Shared/Adapters/DB/`, so this one should be deleted. sorry, it should be caused by my late PR review 🙏

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
LGTM. Thank you for updating package.json!
Don't we write test codes for repository implementation? IMO, it is not necessary or at least can be done later because it doesn't have domain knowledge.
I like the createUserObject is written in repository layer as we talked before. 👍

**Comment by hitohata** [comment]
@taniguchiiqqq 
Thank you for your review.

> Don't we write test codes for repository implementation? IMO, it is not necessary or at least can be done later because it doesn't have domain knowledge.

Yes, that's why I didn't implement tests. This repository has little logic, it's just a wrapper of the NishikiTable Client.

**Comment by taniguchiiqqq** [comment]
@hitohata 
Got it! I'm relieved we are on the same page.

File: backend/main/src/Shared/Adapters/DynamoClient.ts
Comment (hitohata) [reply backend/main/src/Shared/Adapters/DynamoClient.ts:1]: @taniguchiiqqq 
I cannot figure out what file is mentioned. Do you mind if I ask you to specify the exact files?

Comment (taniguchiiqqq) [reply backend/main/src/Shared/Adapters/DynamoClient.ts:1]: @hitohata 
I mean these two files.
`backend/main/src/Shared/Adapters/DynamoClient.ts`
`backend/main/src/Shared/Adapters/DB/DynamoClient.ts`
image

Comment (hitohata) [reply backend/main/src/Shared/Adapters/DynamoClient.ts:1]: @taniguchiiqqq 
Thank you. I got it.
I deleted it.

**Comment by hitohata** [comment]
I resolved the issues you pointed out.
Please check it and approve this PR.

**Review by taniguchiiqqq — APPROVED** [review]

**Pull request merged**